### PR TITLE
ORC: Switch tests to Junit5

### PR DESCRIPTION
**Pull request by skytin1004**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I've switched the files in ORC pakage to JUnit5.

I've checked build successfully.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -957,4 +960,3 @@ apply from: 'jmh.gradle'
 apply from: 'baseline.gradle'
 apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
-
```
Comment (nastra) [inline build.gradle:960]: nit: please revert this change

File: orc/src/test/java/org/apache/iceberg/orc/TestBloomFilter.java [hunk 2]
```diff
@@ -54,12 +52,11 @@ public class TestBloomFilter {
           required(101, "name", Types.StringType.get()),
           required(102, "price", Types.DoubleType.get()));
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File testFile;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestBloomFilter.java:55]: nit: can be made private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataReader.java [hunk 3]
```diff
@@ -43,13 +43,13 @@
...
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 public class TestOrcDataReader implements WithAssertions {
-  @ClassRule public static TemporaryFolder temp = new TemporaryFolder();
+
+  @TempDir static File temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDataReader.java:52]: nit: can probably be made private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java [hunk 4]
```diff
@@ -62,9 +61,9 @@ public class TestOrcDataWriter {
 
   private List<Record> records;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java:64]: private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java [hunk 5]
```diff
@@ -141,7 +137,7 @@ public void testUsingFileIO() throws IOException {
     // local files that would be the LocalFileSystem. To prevent this we use the Proxy classes to
     // use a scheme `dummy` that is not handled. Note that Hadoop 2.7.3 throws IOException
     // while latest Hadoop versions throw UnsupportedFileSystemException (extends IOException)
-    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(temp.newFile()));
+    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(String.valueOf(temp.isFile())));
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java:140]: this seems wrong. This expects a file and not a "true"/"false" string

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java [hunk 6]
```diff
@@ -55,9 +54,9 @@ public class TestOrcDeleteWriters {
 
   private List<Record> records;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:57]: private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java [hunk 7]
```diff
@@ -73,7 +72,7 @@ public void createDeleteRecords() {
 
   @Test
   public void testEqualityDeleteWriter() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:75]: we can probably remove `deleteFile` alltogether

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java [hunk 8]
```diff
@@ -106,12 +104,14 @@ public void testEqualityDeleteWriter() throws IOException {
...
-    Assert.assertEquals("Deleted records should match expected", records, deletedRecords);
+    Assertions.assertThat(deletedRecords)
+        .as("Deleted records should match expected")
+        .isEqualTo(records);
   }
 
   @Test
   public void testPositionDeleteWriter() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:114]: `deleteFile` can be removed

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java [hunk 9]
```diff
@@ -163,13 +162,14 @@ public void testPositionDeleteWriter() throws IOException {
...
-        "Deleted records should match expected", expectedDeleteRecords, deletedRecords);
+    Assertions.assertThat(deletedRecords)
+        .as("Deleted records should match expected")
+        .isEqualTo(expectedDeleteRecords);
   }
 
   @Test
   public void testPositionDeleteWriterWithEmptyRow() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:172]: same here

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 10]
```diff
@@ -50,7 +49,8 @@ public class TestTableProperties {
           Types.NestedField.optional(1, "id", Types.IntegerType.get()),
           Types.NestedField.optional(2, "data", Types.StringType.get()));
 
-  @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+  @TempDir File folder;
+  @TempDir File testFile;
```
Comment (nastra) [inline orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:53]: both can be private

File: build.gradle (hunk 1 above)
Comment (skytin1004) [reply build.gradle]: I've reverted this change

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java (hunk 10 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java]: I've changed 'folder','testFile' to private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java (hunk 8 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java]: I've removed the `deleteFile`

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java (hunk 9 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java]: I've removed the `deleteFile`

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java (hunk 7 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java]: I've removed the deleteFile

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java (hunk 6 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java]: I've changed 'temp' to private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java (hunk 5 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java]: I changed the variable to go inside to fit the context. `String.valueOf(temp.isFile())` to `temp`

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java (hunk 4 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java]: I've changed 'temp' to private

File: orc/src/test/java/org/apache/iceberg/orc/TestOrcDataReader.java (hunk 3 above)
Comment (skytin1004) [reply orc/src/test/java/org/apache/iceberg/orc/TestOrcDataReader.java]: I've changed 'temp' to private

**Comment by nastra** [comment]
@skytin1004 there's a binary file in the PR named [orc/false](https://github.com/apache/iceberg/pull/7954/files#diff-65316bcea8c0ad32a87ef1205a356d75f1eae7fba6be70bdb01b7a286dc5c5bb) that needs to be removed

**Comment by skytin1004** [comment]
Thanks @nastra, I have fixed the code based on your review.

**Comment by skytin1004** [comment]
Thanks @nastra for the detailed review, I have deleted the file [orc/false](https://github.com/apache/iceberg/pull/7954/files#diff-65316bcea8c0ad32a87ef1205a356d75f1eae7fba6be70bdb01b7a286dc5c5bb) and I have modified the code based on your review.

**Review by nastra — APPROVED** [review]
